packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -9,114 +9,224 @@
+      syntheticEvent.stopPropagation();
+      expect(syntheticEvent.isPropagationStopped()).toBe(true);
 
-    expect(nativeEvent.cancelBubble).toBe(true);
-  });
+      expectedCount++;
+    };
+    instance = ReactDOM.render(<div onClick={eventHandler} />, container);
 
-  it('should be able to `persist`', () => {
-    var syntheticEvent = createEvent({});
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    instance.dispatchEvent(event);
 
-    expect(syntheticEvent.isPersistent()).toBe(false);
-    syntheticEvent.persist();
-    expect(syntheticEvent.isPersistent()).toBe(true);
+    expect(expectedCount).toBe(1);
   });
 
-  it('should be nullified if the synthetic event has called destructor and log warnings', () => {
+  it('should be nullified if the synthetic event has called destructor and log warnings`', () => {
     spyOn(console, 'error');
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
-    syntheticEvent.destructor();
-    expect(syntheticEvent.type).toBe(null);
-    expect(syntheticEvent.nativeEvent).toBe(null);
-    expect(syntheticEvent.target).toBe(null);
-    // once for each property accessed
-    expectDev(console.error.calls.count()).toBe(3);
-    // assert the first warning for accessing `type`
-    expectDev(console.error.calls.argsFor(0)[0]).toBe(
-      'Warning: This synthetic event is reused for performance reasons. If ' +
-        "you're seeing this, you're accessing the property `type` on a " +
-        'released/nullified synthetic event. This is set to null. If you must ' +
-        'keep the original synthetic event around, use event.persist(). ' +
-        'See https://fb.me/react-event-pooling for more information.',
-    );
+    var instance;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      syntheticEvent.destructor();
+      expect(syntheticEvent.type).toBe(null);
+      expect(syntheticEvent.nativeEvent).toBe(null);
+      expect(syntheticEvent.target).toBe(null);
+      // once for each property accessed
+      expectDev(console.error.calls.count()).toBe(3);
+      // assert the first warning for accessing `type`
+      expectDev(console.error.calls.argsFor(0)[0]).toBe(
+        'Warning: This synthetic event is reused for performance reasons. If ' +
+          "you're seeing this, you're accessing the property `type` on a " +
+          'released/nullified synthetic event. This is set to null. If you must ' +
+          'keep the original synthetic event around, use event.persist(). ' +
+          'See https://fb.me/react-event-pooling for more information.',
+      );
+
+      expectedCount++;
+    };
+    instance = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    Object.defineProperty(event, 'srcElement', {
```

Comment:
Again, this is specific to IE8 and there is just a single test needed for this. There's no need to add `srcElement` testing to every test.